marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/savedQueries/saveSavedQuery.sjs

```diff
@@ -0,0 +1,36 @@
+'use strict';
+declareUpdate();
+
+var saveQuery;
+var userCollections = ["http://marklogic.com/data-hub/saved-query"];
+var queryDocument = JSON.parse(saveQuery);
+
+try {
+  if(queryDocument.savedQuery.id) {
```

Comment:
In the case of an update, we actually don't want to do a documentInsert. That's because a user may have adjusted the permissions/collections on the document. By doing a documentInsert, we instead have to be careful not to lose any such additional permissions/collections. 

Instead, do this:

    xdmp.nodeReplace(cts.doc("/saved-queries/" + id + ".json"), queryDocument);

That way, you only update the contents of the document, which is all you want to update here.

Reply:
@rahulvudutala If need it, you can use this helper function in hubUtils: https://github.com/marklogic/marklogic-data-hub/blob/develop/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-utils.sjs#L67

Reply:
Added xdmp.nodeReplace